Ensure scripty type for local URLs in importScripts

Kaiido opened the pull request:
See https://github.com/whatwg/html/issues/8869.

Based on https://github.com/web-platform-tests/wpt/blob/master/workers/importscripts_mime.any.js which tests for "external" URLs.

This currently expects the specced behavior, i.e local URLs should also enforce strict MIME types. This fails in both Firefox and Chrome and passes in Safari.  
Note that in its initial form it doesn't use `.any.` because I couldn't find a way to exclude only ServiceWorkers (which don't have access to `createObjectURL`). It also doesn't test for `filesystem:` protocol because I'm not quite sure what's the status of this, and how to write a test for it...

cc @annevk @mikewest

Kaiido committed:
Ensure scripty type for local URLs

See https://github.com/whatwg/html/issues/8869
This currently expects the specced behavior, i.e local URLs should also enforce strict MIME types. This fails in both Firefox and Chrome and passes in Safari.
Note that in its initial form it doesn't use .any. because I couldn't find a way to exclude only ServiceWorkers (which don't have access to createObjectURL). It also doesn't test for `filesystem:` protocol because I'm not quite sure what's the status of this, and to write a test for it.

annevk commented on workers/importscripts_mime_local.worker.js:
```diff
@@ -0,0 +1,68 @@
+  ["TEXT/HTML", false],
+];
+
+for (var test_case of test_cases) {
```
I think this could be a bit clearer:
```suggestion
for (const ([mimeType, isScriptType]) of test_cases) {
```
(Generally we'd also use objects for this reason to express the individual test cases, but I think that's fine here.)

Kaiido replied:
I felt the same when authoring.  
Do you want me to also update [importscripts_mime.any.js](https://github.com/web-platform-tests/wpt/blob/master/workers/importscripts_mime.any.js)? Maybe in another PR would be preferable, or here is fine?

annevk replied:
Yeah feel free to update it as part of this PR.

chrome-dev-stability is known to have timeout issues. If it happens again when you next push we'll ask for admin merge.

annevk reviewed (commented):
Thanks for creating these!

I have a style nit that I think would improve readability quite a bit, but otherwise this looks great.

Kaiido committed:
Use more readable destructuring assignment

As per review comments

Kaiido commented:
Oh, not sure what happened during the checks.  
The changes from the last commit shouldn't have affected it's execution time, and running this test locally I get it done with no noticeable issue.  
Is there a way to perform the checks again maybe?

Kaiido committed:
Use more readable destructuring assignment

Applies the same update as was done in the "local" version of these tests.

annevk commented:
@Kaiido I hadn't fully read OP yet, https://web-platform-tests.org/writing-tests/testharness.html#tests-for-other-or-multiple-globals-any-js shows how you could exclude service workers:

`// META: global=dedicatedworker,window`

would do the trick

Kaiido committed:
Target shared-workers too

Didn't know how to exclude ServiceWorkers, which don't have access to URL.createObjectURL.
Thanks Anne!

annevk commented:
And it seems at this point you could go even further and merge your tests into the other file (just nest your `test()` calls in that for loop at the end).

Kaiido commented:
Though that would prevent the previous tests from running in ServiceWorkers anymore, is that ok?

annevk commented:
Ah okay, you could make that conditional upon the global, but let's just go with this. If it gets confusing for someone later we can always merge then.
